sync2/tokens_table.go

```diff
@@ -42,17 +44,32 @@ func NewTokensTable(db *sqlx.DB, secret string) *TokensTable {
 		-- TODO: FK constraints to devices table?
 		user_id TEXT NOT NULL,
 		device_id TEXT NOT NULL,
+		expired BOOL NOT NULL DEFAULT FALSE,
 		last_seen TIMESTAMP WITH TIME ZONE NOT NULL
 	);`)
 
 	// derive the key from the secret
 	hash := sha256.New()
 	hash.Write([]byte(secret))
 
-	return &TokensTable{
+	t := &TokensTable{
 		db:     db,
 		key256: hash.Sum(nil),
 	}
+
+	var deleteExpiredTokens func()
+	deleteExpiredTokens = func() {
+		deleted, err := t.deleteExpiredTokensAfter(30 * 24 * time.Hour)
```

Comment:
30 days might be a bit much for OIDC, so maybe a bit lower?

Reply:
FWIW I used 30 days in https://github.com/matrix-org/sliding-sync/issues/242 for a different (but related) purpose.